libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1467,21 +1482,29 @@ std::pair<vec_blk_t, bool> PbftManager::comparePbftBlockScheduleWithDAGblocks_(s
     cert_sync_block_.transactions.push_back(*trx);
   }
 
-  auto calculated_order_hash = calculateOrderHash(dag_blocks_order, non_finalized_transactions);
+  const auto calculated_order_hash = calculateOrderHash(dag_blocks_order, non_finalized_transactions);
   if (calculated_order_hash != pbft_block->getOrderHash()) {
-    LOG(log_er_) << "Order hash incorrect. Pbft block: " << pbft_block->getBlockHash()
+    LOG(log_er_) << "Order hash incorrect. Pbft block: " << proposal_block_hash
                  << ". Order hash: " << pbft_block->getOrderHash() << " . Calculated hash:" << calculated_order_hash
                  << ". Dag order: " << dag_blocks_order << ". Trx order: " << non_finalized_transactions;
     dag_blocks_order.clear();
     return std::make_pair(std::move(dag_blocks_order), false);
   }
 
+  // Check reward votes
+  vote_mgr_->updateRewardVotes(proposal_period - 1);
```

Comment:
I dont get why we call it here either ?

Reply:
For example:
1. In PBFT round 10, node1 finalize PBFT block B1 with cert votes {v1, v2, v3} in certify state, then go to next voting stage
2. Node2 running slower than node1 a bit, node2 cert vote v4 for B1 and finalized B1 with cert votes {v1, v2, v3, v4}
3. After go to next round 11, node2 propose a new PBFT block B2 that reward votes are {v1, v2, v3, v4}
4. Node1 need add the v4 to finalized block B1 when node1 want to cert vote on the proposal B2

Reply:
No, we cannot do that.
In above example, 4: 
Node1 cert voting B2 without saving v4 in DB
Node1 restart before finalizing B2, node1 lost v4 in memory.

We have to save all reward votes before we cert voting on the value

Reply:
if Node1 restarts before finalizing B2, he will get B2 after restart together with all rewards votes ?

Reply:
When node1 cert vote on B2, node1 save B2 in DB. Node1 don't need receive B2 from peers again.
In PBFT protocol, cert vote means the node has verified everything and announce to peers. we have to save all states in DB when node trying to cert vote. We should not break the protocol

Reply:
Dont understand. When I cert vote block B2-1 for period 10 and then my node restarts, it does not mean that B2-1 will have enough cert votes to be actually applied, does it ? It can be that some other block B2-2 will be applied for period 10 so once I reconnect to the nodes, my chain size is still 9 and I dont know which pbft block was applied/finalized for period 10 so I must get it somehow ?

Reply:
You comments are right. 
But think about this:
After node1 restart, node1 lost v4 in memory. Then node1 have enough cert votes for B2, node1 will add B2 in PBFT chain. But at the time, node1 may have not sync v4 yet. Node1 will fail to add B2

Reply:
yeah so when it fails to add B2 due to missing v4 - it will get it through syncing little bit later and then can continue in normal operation ?

Reply:
Yes, you are right.
But if keep PBFT states consistent of between restart, that will not fail. This is the reason I prefer saving reward votes in DB when trying to cert vote

Reply:
you have to give me some real argument not just "I think". Because of "you think thats better" we cant really use the simplified design I proposed, which would be much more straightforward and efficient

Reply:
That's better keep PBFT consistent of between node restart